applications/ContactStructuralMechanicsApplication/custom_conditions/mesh_tying_mortar_condition.cpp

```diff
@@ -19,230 +18,245 @@
-const Kratos::Flags MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor>::COMPUTE_RHS_VECTOR(Kratos::Flags::Create(0));
-template< std::size_t TDim, std::size_t TNumNodesElem, TensorValue TTensor>
-const Kratos::Flags MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor>::COMPUTE_LHS_MATRIX(Kratos::Flags::Create(1));
+template< SizeType TDim, SizeType TNumNodesElem, SizeType TNumNodesElemMaster>
+const Kratos::Flags MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster>::COMPUTE_RHS_VECTOR(Kratos::Flags::Create(0));
+template< SizeType TDim, SizeType TNumNodesElem, SizeType TNumNodesElemMaster>
+const Kratos::Flags MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster>::COMPUTE_LHS_MATRIX(Kratos::Flags::Create(1));
 
 /************************************* OPERATIONS **********************************/
 /***********************************************************************************/
 
-template< std::size_t TDim, std::size_t TNumNodesElem, TensorValue TTensor>
-Condition::Pointer MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor>::Create( 
+template< SizeType TDim, SizeType TNumNodesElem, SizeType TNumNodesElemMaster>
+Condition::Pointer MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster>::Create(
     IndexType NewId,
     NodesArrayType const& rThisNodes,
     PropertiesType::Pointer pProperties ) const
 {
-    return Kratos::make_shared< MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor> >( NewId, this->GetGeometry().Create( rThisNodes ), pProperties );
+    return Kratos::make_shared< MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster> >( NewId, this->GetGeometry().Create( rThisNodes ), pProperties );
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
-template< std::size_t TDim, std::size_t TNumNodesElem, TensorValue TTensor>
-Condition::Pointer MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor>::Create(
+template< SizeType TDim, SizeType TNumNodesElem, SizeType TNumNodesElemMaster>
+Condition::Pointer MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster>::Create(
     IndexType NewId,
     GeometryType::Pointer pGeom,
     PropertiesType::Pointer pProperties) const
 {
-    return Kratos::make_shared< MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor> >( NewId, pGeom, pProperties );
+    return Kratos::make_shared< MeshTyingMortarCondition<TDim,TNumNodesElem, TNumNodesElemMaster> >( NewId, pGeom, pProperties );
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
-template< std::size_t TDim, std::size_t TNumNodesElem, TensorValue TTensor>
-Condition::Pointer MeshTyingMortarCondition<TDim,TNumNodesElem,TTensor>::Create(
+template< SizeType TDim, SizeType TNumNodesElem, SizeType TNumNodesElemMaster>
```

Comment:
do you really need to make it all templated? not wrong...just it makes your life way more comlicated

Reply:
Yes, but is the only way to use bounded matrices and vectors